File: c++/swap.cc
#include "swap.h"
#include <iostream>

using namespace std;

void swap::ejecutar(stack<int>& pila){
    int valor1 = pila.top();
    pila.pop();//distinto a java, aquí el popo no returnea, tengo que mirarlo antes de hacer el pop
    int valor2 = pila.top(); 
    pila.pop();
    pila.push(valor1);
    pila.push(valor2);
}

int swap::modificar(int pc){
    return pc + 1;
}

std::string swap::mostrar(){
    return " swap";
}

File: java/swap.java
import java.util.Stack;

class swap extends Instruccion{

    @Override
    public void ejecutar(Stack<Integer> pila) {
        int valor1 = pila.pop();    // desapilo el valor de la cima y lo guardo
        int valor2 = pila.pop();    // desapilo el valor de la cima y lo guardo
        pila.push(valor1);
        pila.push(valor2);  // apilo intercambiando los dos valores que estaban en la cima
    }

    @Override
    public int incrementarPc(int pc) {
        return pc + 1;  // Incremento el contador de programa en 1
    }

    @Override
    public String mostrar() {
        return "    swap";
    }
}
